### ISSUE TITLE
Search fixes

### ISSUE THREAD

**Issue by dominicus** [MutopiaProject/MutopiaWeb#77]
There seems to be something needing fix with search function.
Case1:
From home page, select "Beethoven Piano Sonatas" link from the collections section.  The expected first page in the collection is displayed.  Click on "Next 10", an unrelated set of pieces is shown.  Seems the "next 10" link for collections is not being formed properly.

Case2:
From the home page, do a keyword search for "Pathétique".  I get unrelated results.
Go to "Advanced Search", search for keyword="Pathétique" with no other filters.  I get no results.  Yet we should get at least one hit http://www.mutopiaproject.org/cgibin/piece-info.cgi?id=299

Case3:
From home page, search for "Beethoven".  I get unrelated results.
Repeat keyword search "Beethoven" from "Advanced Search" page, I get expected results.

Case4:
From "advanced search", no keywords, request pieces added/updated in last 3 weeks.  The results include pieces that go beyond the time range.

**Comment by glenl** [comment]
As far as bugs go, this was pretty bad. I have pushed fixes for Case1 and Case3.

I believe Case2 is simply the fact that we do not yet support the full character set. I recall having the discussion in an issue on MutopiaProject in which I posted the code for the fix. I was treading carefully on applying it as it requires some additional overhead at search time.

Ah, and I just found the problem with Case4. Pushing that in now.
